### PR TITLE
Dona/setup dto persistence api call

### PR DESCRIPTION
**Pull request by DonaCrio**
### Abstract
This PR adds an endpoint to the API, as well as the basic layout of the codebase. It also provides a way to run the app locally using docker-compose.

#### Spring API
- `ShowEntity`: Schema for Postgres persistence. This entity is defined using JPA (Java Persistence API), it's the equivalent of Sequelize for NodeJS.
- `ShowDTO`: Data Transfert Object for the `ShowEntity`. This class is non-mutable thus it is used for read-only operations and as to be used as much as possible for data access. It also provides a way of serializing itself into JSON (For API responses).
- `ShowRepository`: JPA repository used to define the operations that can be made through JPA API (Same thing as using an interface to define a contract or behavior for a class)
- `ShowService`: This is the interface that contracts what the service methods will be
- `ShowServiceImpl`: The actual implementation of `ShowService`. It handles all the operations through JPA in transactions.
- `ShowController`: Controller containing the logic of all operations. It uses ShowServiceImpl to handle persisted data and will use another service (that needs to be defined) to handle api calls to Trakt (related to the shows)
- `ShowWebController`: Web controller that defines the entrypoints of the api for the path `/shows`

#### Containers
 You can now launch the app locally by following the README.md

Basically this PR adds a `Dockerfile` to build chowchow-api Spring application and a docker-compose.yml to build all the containers locally.

## Testing Guidelines

You can manually test this PR by following the "Running into Docker container" section of the `README.md` and then perform the following command:
`curl 127.0.0.1:8080/shows/1`

That way you should see a Json response containing a mock show.

## Release Checklist

- [ ] This code has automated tests
- [ ] This code has documentation
- [x] I have a plan to verify that these changes are working as expected after deploy

## Additional Notes
Depends on https://github.com/hinosxz/chowchow/pull/3 for the CI.
Need to setup dev and prod database on heroku first
Need to update this PR to add remote dDBs into env

### CODE REVIEW

File: src/main/java/com/centralesupelec/chowchow/show/service/ShowService.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.centralesupelec.chowchow.show.service;
```
Comment (hinosxz) [inline src/main/java/com/centralesupelec/chowchow/show/service/ShowService.java]: Maybe a bit overkill 🙄
